File: src/main/java/dev/zprestige/fire/module/movement/noslow/EntityUseItemListener.java
package dev.zprestige.fire.module.movement.noslow;

import dev.zprestige.fire.event.bus.EventListener;
import dev.zprestige.fire.event.impl.EntityUseItemEvent;
import net.minecraft.network.play.client.CPacketHeldItemChange;

public class EntityUseItemListener extends EventListener<EntityUseItemEvent, NoSlow> {

    public EntityUseItemListener(final NoSlow noSlow){
        super(EntityUseItemEvent.class, noSlow);
    }

    @Override
    public void invoke(final Object object){
        final EntityUseItemEvent event = (EntityUseItemEvent) object;
        if (event.getEntity().equals(mc.player) && module.slowed() && module.ncpStrict.GetSwitch()){
            mc.player.connection.sendPacket(new CPacketHeldItemChange(mc.player.inventory.currentItem));
        }
    }
}

File: src/main/java/dev/zprestige/fire/module/movement/noslow/EntityUseItemListener.kt
package dev.zprestige.fire.module.movement.noslow

import dev.zprestige.fire.event.bus.EventListener
import dev.zprestige.fire.event.impl.EntityUseItemEvent
import net.minecraft.network.play.client.CPacketHeldItemChange

class EntityUseItemListener(noSlow: NoSlow) : EventListener<EntityUseItemEvent, NoSlow>(EntityUseItemEvent::class.java, noSlow) {
    
    override fun invoke(e: Any) {
        val event = e as EntityUseItemEvent
        if (event.entity == mc.player && module.slowed() && module.ncpStrict.GetSwitch()) {
            mc.player.connection.sendPacket(CPacketHeldItemChange(mc.player.inventory.currentItem))
        }
    }
}